homework/unique_ptr/unique_ptr.hpp

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#include <cstddef>
+
+template <typename T>
+class unique_ptr {
+public:
+    unique_ptr();
+    unique_ptr(std::nullptr_t);
+    explicit unique_ptr(T*);
+    unique_ptr(unique_ptr const&);
+    unique_ptr(unique_ptr&&) noexcept;
+    ~unique_ptr();
+
+    unique_ptr& operator=(unique_ptr const&) = delete;
+    unique_ptr& operator=(std::nullptr_t);
+    unique_ptr& operator=(unique_ptr&&) noexcept;
+    unique_ptr<T>* operator->();
+    T& operator*() const;
+
+    T* get() const;
+    T* release() noexcept;
+    void reset(T*);
```

Comment:
```suggestion
    void reset(T* = nullptr);
```
Default argument will be useful

Reply:
Done